api/src/main/java/org/apache/iceberg/MetadataMaintenance.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+/** APIs for table metadata maintenance, such as removing unused partition specs. */
+public interface MetadataMaintenance {
```

Comment:
Hm I still feel like having the separate Maintenance API has the issue of what goes in here because a user using the API may also feel like expire snapshots should also go here since expire snapshots can also naturally be considered as "metadata maintenance". Maybe it's just a naming issue, so I'll think more about alternatives.

One alternative that comes to mind is `SchemaMaintenance`. Even though we are removing unused specs, the partition spec in the end is how the partition values are derived from the fields in the schema. And it also fits with the future schema pruning we want to do.

I think in my head I was really thinking adding as part of expire snapshots wouldn't add too much complexity to the procedure and users who are running that would also generally want to prune the unused specs/schemas. I'll look into that a bit more.

Reply:
cc @RussellSpitzer @rdblue if they have any thoughts on this